api/STRUCTURE.md

```diff
@@ -0,0 +1,84 @@
+# All of Us Workbench API Structure & Technologies
+
+The _All of Us_ Workbench API is largely built on [Spring Boot](https://spring.io/projects/spring-boot). It also uses [Swagger](https://swagger.io/) for autogenerating REST APIs.
+
+## Database
+
+The primary application database is a [Google Cloud SQL](https://cloud.google.com/sql/) instance running MySQL. We modify its schema using [Liquibase](https://www.liquibase.org/) migrations, which live in [db/changelog/](https://github.com/all-of-us/workbench/tree/master/api/db/changelog).
+
+When creating a new database migration, add an XML file to the changelog directory that describes the migration to perform. This new file should be named `db.changelog-###-description-of-migration.xml`, where ### is the number of the newest migration + 1. Then, add the file to the end of the [db.changelog-master.xml](https://github.com/all-of-us/workbench/blob/master/api/db/changelog/db.changelog-master.xml) file.
+
+We generally name database objects in lower_snake_case.
+
+Liquibase migrations are applied (indirectly) via a bash script called [db/run-migrations.sh](https://github.com/all-of-us/workbench/blob/master/api/db/run-migrations.sh). Do not run this script directly; instead, call one of the following `project.rb` commands:
+
+- dev-up (starts a local dev server)
+- run-local-all-migrations (runs migrations on the `workbench` and `cdr` schemas)
+- run-local-rw-migrations (runs migrations only on the `workbench` schema)
+
+## DAOs
+
+Data Access Objects (DAOs) are interfaces that describe what operations the application can perform on the database. Generaly, we give DAOs the CamelCased name of the database table. All of the Workbench DAOs implement Spring [`CrudRepository`](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html), which means they can do simple operations like find-by-primary-key, save, delete-by-primary-key, etc.
+
+CrudRepository also allows queries to be [automatically derived](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.details) from the name of a method defined on the interface. As such, you will see methods like `findByUserIdAndWorkspaceId` with no apparent implementation. The implementation is generated by Spring.
+
+To define a query that uses a foreign key, there are a couple of options. First, you could use `@JoinColumn`/`@[One|Many]ToOne` annotation on the corresponding DB Model, as described [below](#db-models). Second, you could use an [@Query](https://www.baeldung.com/spring-data-jpa-query) annotation to just describe the query in stringy SQL.
+
+TODO: do we have a preference of which of those to do?
```

Comment:
@ericsong  and I had a pretty good discussion about the various options here this week. I'd love if he could follow-up with some recommendations and/or pointers to models & DAOs where we use healthy patterns.